### ISSUE TITLE
collapse-module-path-index: broke its own contract

### ISSUE THREAD

**Issue by jsmaniac** [racket/racket#1697]
Identifiers produced by `syntax-local-lift-expression` have a module-path-index of `#<module-path-index:()>` as reported by `identifier-binding`. When passing such a MPI to `collapse-module-path-index`, it tries to return `#f` but this return value is not accepted by its contract.

```racket
#lang racket
(require (for-syntax syntax/modcollapse))
(define-syntax (foo stx)
  (displayln
   (collapse-module-path-index
    (car
     (identifier-binding
      (syntax-local-lift-expression
       #'1)))))
  #'(void))

(foo)
```

This program fails with the error:
```
collapse-module-path-index: broke its own contract
  promised: module-path?
  produced: #f
  in: the range of
      (case->
       (-> module-path-index? module-path?)
       [many more cases removed from bug report] …)
```

**Comment by jsmaniac** [comment]
I have a PR ready to update the contract & documentation at https://github.com/jsmaniac/racket/tree/fix-1697-collapse-module-path-index-contract .

Are there any uses of `collapse-module-path-index` within the Racket library which need to be updated to take into account the `#false` case?

**Comment by mflatt** [comment]
Fixed by f459dd9eb7889b1a0b8aaada1e4533951695cc2a, which adjusts the contract
